### PR TITLE
Fix the ClassCastException when running the expiry

### LINKED ISSUE #11775: Map storage incompatible Query parameter
### Describe the bug

The expiration time in the store is a Long, and time return an integer. 

This leads to a ClassCastException in the CriterionOperator.

This exception occurs in the logs after about 15 min when the scheduled Task is started for the first time.
 
```
17:09:43,977 WARN  Timer-0 [CriteriaOperator] Integer
java.lang.ClassCastException: class java.lang.Long cannot be cast to class java.lang.Integer (java.lang.Long and java.lang.Integer are in module java.base of loader 'bootstrap')
	at java.base/java.lang.Integer.compareTo(Integer.java:59)
	at org.keycloak.models.map.storage.chm.CriteriaOperator$ComparisonPredicateImpl.test(CriteriaOperator.java:259)
	at org.keycloak.models.map.storage.chm.MapModelCriteriaBuilder.lambda$fieldCompare$8(MapModelCriteriaBuilder.java:165)
	at java.base/java.util.function.Predicate.lambda$and$0(Predicate.java:69)
	at java.base/java.util.function.Predicate.lambda$and$0(Predicate.java:69)
	at org.keycloak.models.map.storage.chm.ConcurrentHashMapStorage.lambda$delete$1(ConcurrentHashMapStorage.java:119)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:176)
	at java.base/java.util.concurrent.ConcurrentHashMap$EntrySpliterator.forEachRemaining(ConcurrentHashMap.java:3645)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at org.keycloak.models.map.storage.chm.ConcurrentHashMapStorage.delete(ConcurrentHashMapStorage.java:123)
	at org.keycloak.models.map.storage.chm.ConcurrentHashMapKeycloakTransaction$BulkDeleteOperation.execute(ConcurrentHashMapKeycloakTransaction.java:405)
	at org.keycloak.models.map.storage.chm.ConcurrentHashMapKeycloakTransaction.commit(ConcurrentHashMapKeycloakTransaction.java:76)
	at org.keycloak.services.DefaultKeycloakTransactionManager.commit(DefaultKeycloakTransactionManager.java:146)
	at org.keycloak.services.scheduled.ScheduledTaskRunner.runTask(ScheduledTaskRunner.java:75)
	at org.keycloak.services.scheduled.ScheduledTaskRunner.run(ScheduledTaskRunner.java:55)
	at org.keycloak.timer.basic.BasicTimerProvider$1.run(BasicTimerProvider.java:53)
	at java.base/java.util.TimerThread.mainLoop(Timer.java:556)
	at java.base/java.util.TimerThread.run(Timer.java:506)
``` 

### Version

999-SNAPSHOT

### PR DESCRIPTION
**Pull request by ahus1**
Closes #11775

### CODE REVIEW

File: model/map/src/main/java/org/keycloak/models/map/storage/chm/CriteriaOperator.java [hunk 1]
```diff
@@ -261,7 +259,7 @@ public boolean test(Object o) {
             try {
                 return o != null && op.isComparisonTrue(cValue.compareTo(o));
             } catch (ClassCastException ex) {
-                LOG.log(Level.WARNING, "Incomparable argument type for comparison operation: {0}", cValue.getClass().getSimpleName());
+                LOG.log(Logger.Level.WARN, "Incomparable argument type for comparison operation: {0}", cValue.getClass().getSimpleName(), ex);
                 return false;
```
[the remark this answers is not in the document]
Comment (hmlnarik) [reply model/map/src/main/java/org/keycloak/models/map/storage/chm/CriteriaOperator.java:263]: I agree, either the caller needs to fix it or some sort of type conversion needs to happen. Either way, throwing an exception suits better

File: model/map/src/test/java/org/keycloak/models/map/storage/chm/CriteriaOperatorTest.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+package org.keycloak.models.map.storage.chm;
+
+import org.hamcrest.CoreMatchers;
+import org.hamcrest.MatcherAssert;
+import org.junit.Test;
+
+import java.util.function.Predicate;
+
+import static org.junit.Assert.*;
```
Comment (hmlnarik) [inline model/map/src/test/java/org/keycloak/models/map/storage/chm/CriteriaOperatorTest.java:26]: Please avoid star imports

**Review by hmlnarik — CHANGES_REQUESTED** [review]
LGTM, only update the imports please

File: model/map/src/test/java/org/keycloak/models/map/storage/chm/CriteriaOperatorTest.java (hunk 2 above)
Comment (ahus1) [reply model/map/src/test/java/org/keycloak/models/map/storage/chm/CriteriaOperatorTest.java]: Sorry, fixed this.

**Comment by ahus1** [comment]
Test failures seem unrelated to this change, ready for a review.

**Review by hmlnarik — APPROVED** [review]

**Pull request merged**